flang/lib/Lower/ConvertExpr.cpp

```diff
@@ -1149,6 +1178,14 @@ class ScalarExprLowering {
             -> fir::ExtendedValue {
           TODO(loc, "array ref of derived type with length parameters");
         },
+        [&](const Fortran::lower::SymbolBox::IrBox &arr) -> fir::ExtendedValue {
+          // CoordinateOp for IrBoxValue is not generated here. The dimensions
+          // must be kept in the fir.coordinate_op so that potential fir.box
+          // strides can be applied by codegen.
+          mlir::emitError(
+              loc, "internal: IrBoxValue in dim-collapsed fir.coordinate_of");
```

Comment:
Is this a fatal error?

Reply:
Yes, it should be, thanks. I also updated the one just below.